### PR TITLE
[15.0][MIG] `web_advanced_search`: Migration to 15.0

### PR DESCRIPTION
**Pull request by ivantodorovich**
Superseeds https://github.com/OCA/web/pull/2133

### CODE REVIEW

**Comment by ivantodorovich** [comment]
@OCA/community-maintainers it seems pre-commit does not like the new [`/** @odoo-module **/`](https://www.odoo.com/documentation/15.0/developer/reference/frontend/javascript_modules.html#native-javascript-modules) syntax in js files. I'm not even sure what to disable here.. it seems it's just crashing. Any idea how to unblock this?

```
/home/runner/work/web/web/web_advanced_search/static/src/js/AdvancedFilterItem.js
  3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

/home/runner/work/web/web/web_advanced_search/static/src/js/CustomFilterItem.js
  3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

/home/runner/work/web/web/web_advanced_search/static/src/js/FilterMenu.js
  3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

/home/runner/work/web/web/web_advanced_search/static/src/js/RecordPicker.js
  3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
```

**Comment by simahawk** [comment]
> @OCA/community-maintainers it seems pre-commit does not like the new [`/** @odoo-module **/`](https://www.odoo.com/documentation/15.0/developer/reference/frontend/javascript_modules.html#native-javascript-modules) syntax in js files. I'm not even sure what to disable here.. it seems it's just crashing. Any idea how to unblock this?
> 
> ```
> /home/runner/work/web/web/web_advanced_search/static/src/js/AdvancedFilterItem.js
>   3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
> 
> /home/runner/work/web/web/web_advanced_search/static/src/js/CustomFilterItem.js
>   3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
> 
> /home/runner/work/web/web/web_advanced_search/static/src/js/FilterMenu.js
>   3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
> 
> /home/runner/work/web/web/web_advanced_search/static/src/js/RecordPicker.js
>   3:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
> ```

can you open an issue here https://github.com/OCA/oca-addons-repo-template/issues ?

**Comment by ivantodorovich** [comment]
For reference: 
- https://github.com/OCA/oca-addons-repo-template/issues/121

File: .eslintrc.yml
Comment (ivantodorovich) [inline .eslintrc.yml:7]: - https://github.com/OCA/oca-addons-repo-template/pull/122

**Comment by onurugur** [comment]
@ivantodorovich when pressing the  filter
```
UncaughtPromiseError
Uncaught Promise > Cannot find the definition of component "AdvancedFilterItem"

Error: Cannot find the definition of component "AdvancedFilterItem"
    at QWeb.eval (eval at _compile (http://localhost:9090/web/assets/debug/web.assets_common.js:10280:28), <anonymous>:140:27) (/web/static/lib/owl/owl.js:1795)
    at QWeb.eval (eval at _compile (http://localhost:9090/web/assets/debug/web.assets_common.js:10280:28), <anonymous>:58:21) (/web/static/lib/owl/owl.js:1795)
    at QWeb.render (http://localhost:9090/web/assets/debug/web.assets_common.js:10201:32) (/web/static/lib/owl/owl.js:1716)
    at Dropdown.__render (http://localhost:9090/web/assets/debug/web.assets_common.js:13055:37) (/web/static/lib/owl/owl.js:4570)
    at http://localhost:9090/web/assets/debug/web.assets_common.js:12800:26 (/web/static/lib/owl/owl.js:4315) 

```
also in console

```
2VM3596:140 Uncaught (in promise) Error: Cannot find the definition of component "AdvancedFilterItem"
    at QWeb.eval (eval at _compile (owl.js:1795:1), <anonymous>:140:27)
    at QWeb.eval (eval at _compile (owl.js:1795:1), <anonymous>:58:21)
    at QWeb.render (owl.js:1716:1)
    at Dropdown.__render (owl.js:4570:1)
    at owl.js:4315:1
   
```
   
   Would you check these?

**Comment by ivantodorovich** [comment]
> @ivantodorovich when pressing the filter
> 
> Would you check these?

Sorry about this. It's fixed now.
I forgot to rename the imports after renaming the files to have pre-commit pass 😅

**Comment by onurugur** [comment]
Thank you. It working. Well done.

**Review by celm1990 — APPROVED** [review]
functional testing OK

**Comment by pedrobaeza** [comment]
A missing gap that I have always missed is that on advanced filter, on a many2one, there's no option of searching "is equal to" but the many2one widget. Only to put the exact text or the ID.

**Comment by ivantodorovich** [comment]
> A missing gap that I have always missed is that on advanced filter, on a many2one, there's no option of searching "is equal to" but the many2one widget. Only to put the exact text or the ID.

What do you mean? 🤔

**Comment by pedrobaeza** [comment]
I mean to have this:

Selección_016

here:

Selección_017

**Comment by HaraldPanten** [comment]
Hi, any news about this PR?

THX!

**Review by flachica — APPROVED** [review]

**Comment by ghost** [comment]
@Yajo can you please merge this PR?

**Pull request merged**
